backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/MoveDiskCommand.java

```diff
@@ -170,11 +170,15 @@ protected Map<String, Pair<String, String>> getExclusiveLocks() {
 
     private String getDiskIsBeingMigratedMessage() {
         if (cachedDiskIsBeingMigratedMessage == null) {
-            cachedDiskIsBeingMigratedMessage = new LockMessage(EngineMessage.ACTION_TYPE_FAILED_DISK_IS_BEING_MIGRATED)
-                    .withOptional("DiskName", getParameters().getNewAlias())
-                    .toString();
+            if (getParameters().getNewAlias() == null && getParameters().getDiskAlias() == null) {
+                cachedDiskIsBeingMigratedMessage = new LockMessage(EngineMessage.ACTION_TYPE_FAILED_OBJECT_LOCKED).toString();
+            } else {
+                cachedDiskIsBeingMigratedMessage = new LockMessage(EngineMessage.ACTION_TYPE_FAILED_DISK_IS_BEING_MIGRATED)
+                        .withOptional("DiskName", getParameters().getNewAlias() != null ?
+                                getParameters().getNewAlias() : getParameters().getDiskAlias())
```

Comment:
isn't this enough?
it makes sense that we won't have NewAlias when moving a disk, but DiskAlias should always be available

Reply:
right and even if let's say we don't have DiskAlias, I'd say it's better to indicate the fact that the other operation is blocked because disk is being migrated (without alias) than having the general "Related operation is currently in progress" message

Reply:
That way, the only thing that is being changed among the messages is the value in DiskName.
lines 174-176 and 178-180 are duplicated.
How about using a disk name parameter set to "" at first, setting its value according to the 'if' in lines 179-181, and to use the lockMessage with the disk name parameter?